lib/dynamic_screening_solutions/api.py

```diff
@@ -108,6 +108,20 @@ def handle_bad_response(self, response):
             extra_data=extra_data
         )
 
+    def handle_response(self, response):
+        try:
+            response_json = response.json()
+            response_json = None if type(response_json) == dict and response_json.get('error') else response_json
+        except ValueError:
+            response_json = None
+
+        if response_json is None:
+            self.handle_bad_response(response)
```

Comment:
Needs a `handle_error_response` method for handling an actual 200 response with `{‘error’ : … }` in the response as valid JSON, rather than 1 single `handle_bad_response` (non-valid JSON returned).